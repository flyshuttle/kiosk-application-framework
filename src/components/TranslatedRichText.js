import { createElement } from "react";
import parse from "react-html-parser";

import "../i18n";
import { useTranslation } from "react-i18next";

export default function TranslateRichText(props) {
  const { t } = useTranslation();

  const tag = props.wrappingTag || "div";
  let elementProps = {};
  const contentKey = props.children || "no key was set";

  if (props.className !== undefined) {
    elementProps = { className: props.className };
  }
  // const elementProps =
  const parsedContent = parse(t(contentKey));

  if (props.wrappingTag === undefined) {
    return parsedContent;
  } else {
    return createElement(tag, elementProps, parsedContent);
  }
}
