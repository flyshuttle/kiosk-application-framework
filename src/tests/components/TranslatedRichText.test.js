import React from "react";
import ReactDOM from "react-dom";
import { create } from "react-test-renderer";

import TranslatedRichText from "../../components/TranslatedRichText";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<TranslatedRichText />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the component with no props or content", () => {
  const component = create(<TranslatedRichText />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders the component with a specified span type", () => {
  const component = create(<TranslatedRichText tagType="span" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders the component with a specified span type and class", () => {
  const component = create(
    <TranslatedRichText tagType="span" className="css-class" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
