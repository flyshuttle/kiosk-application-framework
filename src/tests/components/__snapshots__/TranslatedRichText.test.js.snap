// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component with a specified span type 1`] = `"no key was set"`;

exports[`renders the component with a specified span type and class 1`] = `"no key was set"`;

exports[`renders the component with no props or content 1`] = `"no key was set"`;
